### PR TITLE
Add doc for spacemacs/restart-emacs keybinding

### LINKED ISSUE #4397: `SPC q r` should be documented in doc/DOCUMENTATION.org
`SPC q r` (`spacemacs/restart-emacs`) is a pretty useful command.  IMHO it should be documented in `doc/DOCUMENTATION.org`.

### PR DESCRIPTION
**Pull request by balajisivaraman**
Closes #4397

### CODE REVIEW

File: doc/DOCUMENTATION.org [hunk 1]
```diff
@@ -2658,12 +2658,13 @@ variable =dotspacemacs-persistent-server= to =t= in your =~./spacemacs=.
...
-|------------+--------------------------------------------------------------------|
-| ~SPC q q~  | Quit Emacs and kill the server, prompt for changed buffers to save |
-| ~SPC q Q~  | Quit Emacs and kill the server, lose all unsaved changes.          |
-| ~SPC q s~  | Save the buffers, quit Emacs and kill the server                   |
-| ~SPC q z~  | Kill the current frame                                             |
+| Keybinding | Description                                                              |
+|------------+--------------------------------------------------------------------------|
+| ~SPC q q~  | Quit Emacs and kill the server, prompt for changed buffers to save       |
+| ~SPC q r~  | Restart both Emacs and the server, prompting to save any changed buffers |
+| ~SPC q Q~  | Quit Emacs and kill the server, lose all unsaved changes.                |
```
Comment (StreakyCobra) [inline doc/DOCUMENTATION.org]: We try to keep keybindings sorted by alphabetical order :-) `q -> Q -> r -> …`

Comment (balajisivaraman) [inline doc/DOCUMENTATION.org]: @StreakyCobra, Oh OK! Will update the PR. :-)

**Comment by StreakyCobra** [comment]
:+1:

**Comment by syl20bnr** [comment]
Thank you, nice doc fixes :+1: 
Released in 0.105.9

**Pull request closed without merging**